parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnWriterV1.java

```diff
@@ -109,9 +114,11 @@ private void accountForValueWritten() {
           + dataColumn.getBufferedSize();
       if (memSize > pageSizeThreshold) {
         // we will write the current page and check again the size at the predicted middle of next page
-        valueCountForNextSizeCheck = valueCount / 2;
+        if(estimateNextSizeCheck) {
```

Comment:
I believe that if `estimateNextSizeCheck` is false, it will always use `valueCountForNextSizeCheck` for the next check which is set to `initialRowCountForSizeCheck`, so I believe it does exactly what you want it to do (i.e. always check at the initial interval without doing an estimate).